packages/data-stores/src/site/test/reducer.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * These tests shouldn't require the jsdom environment,
+ * but we're waiting for this PR to merge:
+ * https://github.com/WordPress/gutenberg/pull/20486
+ *
+ * @jest-environment jsdom
+ */
+
+/**
+ * Internal dependencies
+ */
+import { existingSite } from '../reducer';
+import { createActions } from '../actions';
+import { ExistingSiteDetails, ExistingSiteError } from '../types';
+
+const { receiveExistingSite, receiveExistingSiteFailed } = createActions( {
+	client_id: '',
+	client_secret: '',
+} );
+
+describe( 'existingSite', () => {
+	const existingSiteDetails: ExistingSiteDetails = {
+		ID: 1,
+		name: 'My test site',
+		description: '',
+		URL: 'http://mytestsite12345.wordpress.com',
+	};
+
+	const existingSiteError: ExistingSiteError = {
+		error: 'unknown_blog',
+		message: 'Unknown blog',
+	};
+
+	it( 'returns the correct default state', () => {
+		const state = existingSite( undefined, { type: 'TEST_ACTION' } );
+		expect( state ).toEqual( {} );
+	} );
+
+	it( 'returns site data keyed by slug', () => {
+		const state = existingSite(
+			undefined,
+			receiveExistingSite( 'mytestsite12345.wordpress.com', existingSiteDetails )
```

Comment:
IMO this should be explicit rather than using `receiveExistingSite`, since right now, it's not technically a unit test for `existingSite` but implicitly relies on `receiveExistingSite` doing the right thing (and if it doesn't, it'd break, but point at `existingSite` as the culprit).

Reply:
Updated to be explicit 👍